standalone/app/controllers/OAuthLoginController.scala

```diff
@@ -57,13 +57,20 @@ object OAuthLoginController extends Controller with ExecutionContexts with impli
     }
   }
 
+  private def overrideRedirectUrl(config: GoogleAuthConfig)(implicit request: RequestHeader) = {
+    request.getQueryString("redirect-url") match {
+      case Some(url) => config.copy(redirectUrl = url)
```

Comment:
Yeah we could. The redirect URLs are whitelisted in the Google apps account settings globally, but an extra check on domain here should be fine.